src/Database/Postgres/Temp/Internal/Core.hs

```diff
@@ -213,12 +213,16 @@ executeProcessAndTee
   -- ^ Process config
   -> IO (ExitCode, String, String)
 executeProcessAndTee name config = fmap (\((x, y), z) -> (x, z, y)) $
-  teeHandle (completeProcessConfigStdOut config) $ \newOut ->
-    teeHandle (completeProcessConfigStdErr config) $ \newErr ->
-      executeProcess name $ config
+  teeHandle (completeProcessConfigStdOut config) $ \newOut aOut ->
+    teeHandle (completeProcessConfigStdErr config) $ \newErr aErr -> do
+      result <- executeProcess name $ config
         { completeProcessConfigStdErr = newErr
         , completeProcessConfigStdOut = newOut
         }
+      for_ [newOut, newErr] hClose -- executeProcess doesn't close these
```

Comment:
`executeProcess` doesn't close `newErr` or `newOut` because it uses `createProcess_` internally, so we explicitly do that now. This causes the background threads to die eventually when they try to `hGetLine` on a handle that's hit EOF.

Reply:
One thing I wasn't totally sure about: we close the write ends here (so the read threads die), but we also close the write ends in the cleanup function of the bracketed `createPipe`. So, the handles are closed twice. Things seem to work, though; I guess I'd have expected that to be an IO exception.

Reply:
Oh, scratch that. From the docs:

> Performing hClose on a handle that has already been closed has no effect; doing so is not an error.